src/libmongoc/tests/unified/result.c

```diff
@@ -565,6 +567,43 @@ result_check (result_t *result,
             goto done;
          }
       }
+
+      if (error_response) {
```

Comment:
This isn't something that should block merging, but it looks like this function has grown to be pretty long. I wonder if we should start thinking of breaking it up into smaller pieces?